code/modules/reagents/chemistry/reagents/other_reagents.dm

```diff
@@ -1736,3 +1736,44 @@
 	color = "#ED2939"
 	taste_description = "upside down"
 	can_synth = FALSE
+
+/// Improvised reagent that induces vomiting. Created by dipping a dead mouse in welder fluid.
+/datum/reagent/yuck
+	name = "Organic Slurry"
+	description = "A mixture of various colors of fluid. Induces vomiting."
+	glass_name = "glass of ...yuck!"
+	glass_desc = "It smells like a carcass, and doesn't look much better."
+	color = "#545000"
+	taste_description = "insides"
+	taste_mult = 4
+	can_synth = FALSE
+	metabolization_rate = 0.4 * REAGENTS_METABOLISM
+	var/yuck_cycle = 0 //! The `current_cycle` when puking starts.
+
+/datum/reagent/yuck/on_mob_add(mob/living/L)
+	if(HAS_TRAIT(src, TRAIT_NOHUNGER)) //they can't puke
+		holder.del_reagent(type)
+
+#define PUKE_CYCLES 3 		// every X cycle is a puke
+#define PUKES_TO_STUN 3 	// hit this amount of pukes in a row to start stunning
+/datum/reagent/yuck/on_mob_life(mob/living/carbon/C)
+	if(!yuck_cycle)
+		if(prob(8))
+			var/dread = pick("Something is moving in your stomach...", \
+				"A wet growl echoes from your stomach...", \
+				"For a moment you feel like [get_area(C)] is moving, but it's your stomach...")
```

Comment:
Wouldn't this have some weird sentences like "...you feel like Kitchen is moving...", "...you feel like Bar is moving..." etc? Should probably have "the" in front of the get_area.

Reply:
I originally had "the station" but you might be in space when this happens. It'd be nice to be able to go back to a generic like that because some areas like Kitchen need the, and some like Surgery don't. "the ground" might work?